### PR TITLE
add "Friends App"

### PR DESCRIPTION
**Pull request by SergSenras**
### Friends App

[Demo](https://sergsenras.github.io/friends-app/) |
[Code base](https://github.com/SergSenras/SergSenras.github.io/tree/master/friends-app)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/SergSenras/friends-app/script.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    }
+    else if(btn){
+      displayList = persons;
+    }
+    displayCards();
+  })
+
+  SEARCH_FIELD.addEventListener('keyup', function(event){
+    searchStr = SEARCH_FIELD.value;
+    displayList = persons.filter(el => el.name.last.indexOf(searchStr) != -1 || el.name.first.indexOf(searchStr) != -1)
```
Comment (AMashoshyna) [inline submissions/SergSenras/friends-app/script.js]: Take a look at [String.prototype.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes)

File: submissions/SergSenras/friends-app/script.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          break;
+      }
+    }
+    else if(btn){
+      displayList = persons;
+    }
+    displayCards();
+  })
+
+  SEARCH_FIELD.addEventListener('keyup', function(event){
```
Comment (AMashoshyna) [inline submissions/SergSenras/friends-app/script.js]: What if I paste the search phrase with Ctrl+V?

File: submissions/SergSenras/friends-app/script.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
+const CONTAINER = document.getElementById('container');
+const SIDEBAR = document.getElementById('sidebar');
+const SEARCH_FIELD = document.getElementById('searchField');
+const URL = 'https://randomuser.me/api/?results=40';
+
+let persons = [];
+let displayList = [];
+let searchStr = '';
```
Comment (AMashoshyna) [inline submissions/SergSenras/friends-app/script.js]: Are you sure you need to make `searchStr` global variable?

File: submissions/SergSenras/friends-app/script.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+async function loadXMLDoc() {
```
Comment (AMashoshyna) [inline submissions/SergSenras/friends-app/script.js]: Could you please explain the choice of the name for this function? XML is a pretty specific name of a format.

File: submissions/SergSenras/friends-app/script.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  let response = await fetch(URL)
```
Comment (AMashoshyna) [inline submissions/SergSenras/friends-app/script.js]: It is great that you have learned async functions, but in this particular case it does not seem to be a good use case for async/await. Your function doesn't return result of API query, so the code after this function call will run immediately. Usually async/await are most useful when you need to make several asynchronous operations sequentially and you don't want to use promise chains (they may look confusing indeed).

**Comment by AMashoshyna** [comment]
Great job! Let's make some improvements.

User interface: if I enter empty string, search will be performed. Usually you should trim the input (remove whitespaces at the beginning and in the end of the string) and check for empty input.

Code formatting: please make code formatting (indentations and empty lines in particular) consistent. You can use your code editor's built-in formatting tools.

File: submissions/SergSenras/friends-app/script.js (hunk 1 above)
Comment (SergSenras) [reply submissions/SergSenras/friends-app/script.js]:
```suggestion
	SEARCH_FIELD.addEventListener('input', function (event) {
		let searchStr = '';
		searchStr = SEARCH_FIELD.value.trim();
		displayList = persons.filter(el => el.name.last.includes(searchStr));
```

File: submissions/SergSenras/friends-app/script.js (hunk 2 above)
Comment (SergSenras) [reply submissions/SergSenras/friends-app/script.js]:
```suggestion
  SEARCH_FIELD.addEventListener('input', function (event) {
```

File: submissions/SergSenras/friends-app/script.js (hunk 3 above)
Comment (SergSenras) [reply submissions/SergSenras/friends-app/script.js]: At first planned to use it in different places, that's why made global.
Fixed

File: submissions/SergSenras/friends-app/script.js (hunk 4 above)
Comment (SergSenras) [reply submissions/SergSenras/friends-app/script.js]: Well, that was the deadline )))
Fixed
```suggestion
function getPeopleData() {
```

File: submissions/SergSenras/friends-app/script.js (hunk 5 above)
Comment (SergSenras) [reply submissions/SergSenras/friends-app/script.js]: "Your function doesn't return result of API query" ©
why not?

Thanks for the explanations!
Fixed

**Comment by SergSenras** [comment]
@AMashoshyna Made all requested chenges. Please review when you have a spare moment )

File: submissions/SergSenras/friends-app/script.js (hunk 5 above)
Comment (AMashoshyna) [reply submissions/SergSenras/friends-app/script.js]: The return value of `loadXMLDoc` is a Promise, but the value of this Promise is `undefined`. You put the result of API request into `response` variable, but you don't return it. It works for this particular case, but a more traditional approach for Promise chain is to return the result of the asynchronous action into next callback. Here is an article that helped me a lot: [We have a Problem with Promises](https://pouchdb.com/2015/05/18/we-have-a-problem-with-promises.html)

**Comment by AMashoshyna** [comment]
Well done!

File: submissions/SergSenras/friends-app/script.js (hunk 5 above)
Comment (SergSenras) [reply submissions/SergSenras/friends-app/script.js]: @AMashoshyna Thanks a lot!

**Pull request merged**